Incorrect typing of `TextDecoderStream`

### What version of Bun is running?

1.2.17+282dda62c

### What platform is your computer?

Darwin 24.5.0 arm64 arm

### What steps can reproduce the bug?

Run `bun init` in an empty dir

Paste the following into `index.ts`:

```ts
const stream = Bun.file('./foo.csv')
  .stream()
  .pipeThrough(new TextDecoderStream())
```

### What is the expected behavior?

No type errors

### What do you see instead?

```
index.ts:3:16 - error TS2345: Argument of type 'TextDecoderStream' is not assignable to parameter of type 'ReadableWritablePair<unknown, Uint8Array<ArrayBufferLike>>'.
  Type '{}' is missing the following properties from type 'ReadableWritablePair<unknown, Uint8Array<ArrayBufferLike>>': readable, writable

3   .pipeThrough(new TextDecoderStream())
                 ~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in index.ts:3
```

### Additional information

_No response_

--- a/packages/bun-types/globals.d.ts
+++ b/packages/bun-types/globals.d.ts
@@ -15,8 +15,14 @@ declare module "bun" {
     type LibEmptyOrBunWebSocket = LibDomIsLoaded extends true ? {} : Bun.WebSocket;
 
     type LibEmptyOrNodeUtilTextEncoder = LibDomIsLoaded extends true ? {} : import("node:util").TextEncoder;
+    type LibEmptyOrNodeStreamWebTextEncoderStream = LibDomIsLoaded extends true
+      ? {}
+      : import("node:stream/web").TextEncoderStream;
 
     type LibEmptyOrNodeUtilTextDecoder = LibDomIsLoaded extends true ? {} : import("node:util").TextDecoder;
+    type LibEmptyOrNodeStreamWebTextDecoderStream = LibDomIsLoaded extends true
+      ? {}
+      : import("node:stream/web").TextDecoderStream;
 
     type LibEmptyOrNodeReadableStream<T> = LibDomIsLoaded extends true
       ? {}
@@ -1656,13 +1662,13 @@ declare var ReadableStreamBYOBRequest: Bun.__internal.UseLibDomIfAvailable<
   { prototype: ReadableStreamBYOBRequest; new (): ReadableStreamBYOBRequest }
 >;
 
-interface TextDecoderStream {}
+interface TextDecoderStream extends Bun.__internal.LibEmptyOrNodeStreamWebTextDecoderStream {}
 declare var TextDecoderStream: Bun.__internal.UseLibDomIfAvailable<
   "TextDecoderStream",
   { prototype: TextDecoderStream; new (): TextDecoderStream }
 >;
 
-interface TextEncoderStream {}
+interface TextEncoderStream extends Bun.__internal.LibEmptyOrNodeStreamWebTextEncoderStream {}
 declare var TextEncoderStream: Bun.__internal.UseLibDomIfAvailable<
   "TextEncoderStream",
   { prototype: TextEncoderStream; new (): TextEncoderStream }

--- a/test/integration/bun-types/fixture/streams.ts
+++ b/test/integration/bun-types/fixture/streams.ts
@@ -45,3 +45,16 @@ expectType(stream.json()).is<Promise<any>>();
 expectType(stream.bytes()).is<Promise<Uint8Array>>();
 expectType(stream.text()).is<Promise<string>>();
 expectType(stream.blob()).is<Promise<Blob>>();
+
+Bun.file("./foo.csv").stream().pipeThrough(new TextDecoderStream()).pipeThrough(new TextEncoderStream());
+
+Bun.file("./foo.csv")
+  .stream()
+  .pipeThrough(new TextDecoderStream())
+  .pipeTo(
+    new WritableStream({
+      write(chunk) {
+        expectType(chunk).is<string>();
+      },
+    }),
+  );

